### ISSUE TITLE
Normalization in forecasting dataset

### ISSUE THREAD

**Issue by nnsriram97** [argoai/argoverse-api#41]
Hey thanks a lot for the dataset. Looking forward for your baseline codes. I would like to clarify on how you had normalized the trajectories for training. The paper mentions that the trajectory starts at origin for all the AGENTS and goes in the positive x direction. I would like to know how you had aligned these trajectories? Also, how is y at T_obs = 0? Kindly clarify. Thank you once again!

**Comment by James-Hays** [comment]
Hi Sriram,

At a high level, this is a normalization of 2d position and rotation (not scale). So there are three parameters. 

First, an intuition behind this normalization. We're assuming absolute position in the map doesn't matter. A trajectory at (300, 300) should behave like one at (500, 500). This isn't strictly true. Different parts of the map have different trajectory distributions. But the training / val / and test sets are geographically distinct so the dataset is forcing you to learn to generalize to new pieces of road with the current splits. It would be interesting to formulate the splits with geographic overlap to see if you can "overfit" to particular locations, though. Anyway, this location invariance assumption is why we move trajectories to start at (0,0).

Next, we assume that absolute orientation doesn't matter. Trajectories headed south don't really differ from trajectories headed west. Thus each trajectory is rotated so that the last observation is on the positive x axis (that is equivalent to the y value at T_obs being 0). Again, this assumption of invariance could be wrong. Especially at a particular location, people driving south may really behave differently than people driving west.

**Comment by nnsriram97** [comment]
Hi James,

Thanks a lot for the information. I understand the need for translation and rotational invariance in the data for training. Although, I have a doubt how a translated and rotated trajectory starting at the origin (0,0) can have y at T_obs = 0 i.e. (some x, 0)? (If this is what you're mentioning)

I believe you're rotating the trajectories based on the lane direction that the vehicle is at? Also, it takes a very long time and gets stuck while trying to access the get_lane_direction function for the train data. Kindly take a look into it.

**Comment by jagjeet-singh** [comment]
Hi Sriram,
Think of translation and rotation as two steps. First, you translate the trajectory as is such that it starts at the origin. The trajectory at this point can be pointing in any direction. So you then compute the rotation angle such that trajectory at `T_obs` crosses the x-axis. Once you have this angle, just rotate it and you will have a trajectory which starts at (0,0) and has `y=0` at `T_obs`.
Hope this helps.
